plugins/controlboard/src/ControlBoard.cpp

```diff
@@ -0,0 +1,353 @@
+    yDebug() << "gz-sim-yarp-controlboard-system : device scoped name: " << m_deviceScopedName;
+
+    m_modelEntity = _entity;
+
+    m_controlBoardData.modelScopedName = m_robotScopedName;
+
+    m_pluginParameters.put(yarp::dev::gzyarp::YarpControlBoardScopedName.c_str(),
+                           m_robotScopedName.c_str());
+
+    // Insert the pointer in the singleton handler for retrieving it in the yarp driver
+    ControlBoardDataSingleton::getControlBoardHandler()->setControlBoardData(&(m_controlBoardData));
+
+    m_pluginParameters.put("device", "gazebo_controlboard");
+    m_pluginParameters.put("name", m_deviceScopedName);
+    m_pluginParameters.put("robotScopedName", m_robotScopedName);
+
+    if (_sdf->HasElement("initialConfiguration"))
+    {
+        std::string initialConfiguration = _sdf->Get<std::string>("initialConfiguration");
+        yDebug() << "gz-sim-yarp-controlboard-system: initialConfiguration: "
+                 << initialConfiguration;
+        m_pluginParameters.put("initialConfiguration", initialConfiguration);
+    }
+
+    if (!m_controlBoardDriver.open(m_pluginParameters))
+    {
+        yError() << "gz-sim-yarp-controlboard-system Plugin failed: error in opening yarp "
+                    "driver";
+        return;
+    }
+
+    if (!Handler::getHandler()->setDevice(m_deviceScopedName, &m_controlBoardDriver))
+    {
+        yError() << "gz-sim-yarp-basestate-system: failed setting scopedDeviceName(="
+                 << m_deviceScopedName << ")";
+        return;
+    }
+
+    if (!setJointProperties(_ecm))
+    {
+        yError() << "gz-sim-yarp-controlboard-system: failed setting joint properties";
+        return;
+    }
+
+    yInfo() << "Registered YARP device with instance name:" << m_deviceScopedName;
+    m_deviceRegistered = true;
+}
+
+void ControlBoard::PreUpdate(const UpdateInfo& _info, EntityComponentManager& _ecm)
+{
+    if (!updateReferences(_ecm))
+    {
+        yError() << "Error while updating control references";
+    }
+}
+
+void ControlBoard::PostUpdate(const UpdateInfo& _info, const EntityComponentManager& _ecm)
+{
+    updateSimTime(_info);
+
+    if (!readJointsMeasurements(_ecm))
+    {
+        yError() << "Error while reading joints measurements";
+    }
+
+    // checkForJointsHwFault();
+}
+
+void ControlBoard::Reset(const UpdateInfo& _info, EntityComponentManager& _ecm)
+{
+    // TODO
+}
+
+// Private methods
+
+bool ControlBoard::setJointProperties(EntityComponentManager& _ecm)
+{
+    yarp::os::Bottle jointsFromConfig = m_pluginParameters.findGroup("jointNames");
+
+    if (jointsFromConfig.isNull())
+    {
+        yError() << "Error while reading jointNames from plugin parameters";
+        return false;
+    }
+
+    auto jointsFromConfigNum = jointsFromConfig.size() - 1; // -1 to exclude the group name
+    yInfo() << "Found " + std::to_string(jointsFromConfigNum)
+                   + " joints from the plugin configuration.";
```

Comment:
I think the output of the plugin become messy soon (as it is in gazebo-yarp-plugins). If this is not needed, I would delete or comment it.

Reply:
Ok, maybe this time I will just comment this kind of logs since they are useful for debugging and rewriting it every time is tedious. Ideally, they could be debug or trace level logs in the future, but I don't know how to set the log level when running the plugin at the moment.

Reply:
I do not think YARP logger supports that. We can consider switching using gz's logging infrastructure instead. However, we can decide on this in the future.